site/app/models/CalendarInfo.php

```diff
@@ -14,6 +14,7 @@
  * Container of information used to fill the calendar.
  */
 class CalendarInfo extends AbstractModel {
+    const COLORS = ['#c3a2d2','#99b270','#cd98aa','#6bb88f','#c8938d','#6b9fb8','#c39e83','#98a3cd','#8ac78e','#b39b61','#6eb9aa','#b4be79','#94a2cc','#80be79','#b48b64','#b9b26e','#83a0c3','#ada5d4','#e57fcf','#c0c246'];
```

Comment:
This line is very long and should be reformatted.

Should the colors be listed here anyway?  Why not do this kind of logic in the relevant view?  No need to clutter the model with this kind of UI business.

Additionally, where did all of these colors come from?  Is there a way to share code with another part of the codebase somehow?

Reply:
I honestly don't know what this code does either. It is referenced once from what I can tell (basically these are the colors that the gradeables are given for each course). When I commented it out and created new gradeables though, it didn't actually break anything (which really surprised me). As for where the colors came from, this is before my time so I can only speculate but my hunch is that these are colors that Chris (and probably other people) agreed are distinct enough to be the valid colors for gradeables. If you have any idea what this does please reach out but I'll see if I can find other ways to get colors for gradeables.

Reply:
I believe I took these colors from office hours. I didn't worry about sharing the code at the time since this was still a draft PR but that would probably be good to do.

Reply:
I'll take a look and see how I can edit this. Basically the colors gets assigned in here so that it can be accessed through each course. I don't know if this is the best way to do it. As for how the colors came to be, it is before my time. My guess would be that Chris probably got input from people who decided on these colors. Not totally sure.